backend/validators/llm.py

```
@@ -149,3 +153,56 @@ async def provider_available(
+                ],
+            }
+
+            api_key = os.environ.get(key_env)
+            if not api_key:
+                print(f"ERROR: missing API key for {key_env}")
+                return False
+
+            async with aiohttp.ClientSession() as session:
+                async with session.post(
+                    url, json=prompt, headers={"Authorization": f"Bearer {api_key}"}
+                ) as response:
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)